src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -4,161 +4,219 @@
-    if (m_transform.first == TransformType::PERMUTE) {
-        const auto& axes = std::get<std::vector<std::size_t>>(m_transform.second);
-        for (const auto& axis : axes) {
-            seed ^= std::hash<std::size_t>()(axis) + 0x9e3779b9;
-        }
-    } else if (m_transform.first == TransformType::CONCAT) {
-        const auto& axis = std::get<ConcatMeta>(m_transform.second).second;
-        seed ^= std::hash<std::size_t>()(axis) + 0x9e3779b9;
-        for (auto& lt : std::get<ConcatMeta>(m_transform.second).first) {
-            seed ^= lt.get_hash() + 0x9e3779b9;
-        }
-    } else if (m_transform.first == TransformType::UNPACK) {
-        const auto& unpack_meta = std::get<UnpackMeta>(m_transform.second);
-        seed ^= std::get<0>(unpack_meta).get_hash() + 0x9e3779b9;
-        seed ^= std::get<1>(unpack_meta).get_hash() + 0x9e3779b9;
-        seed ^= std::get<2>(unpack_meta).get_hash() + 0x9e3779b9;
-        for (const auto& dim : std::get<3>(unpack_meta)) {
-            seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
-        }
-        seed ^= std::get<4>(unpack_meta).hash() + 0x9e3779b9;
-    }
+    std::visit(overloaded{[](const auto& op) { /* do nothing */ },
+                          [&seed](const op::Permute& op) {
+                              for (const auto& axis : op.axes) {
+                                  seed ^= std::hash<std::size_t>()(axis) + 0x9e3779b9;
+                              }
+                              seed ^= op.tensor.get_hash() + 0x9e3779b9;
+                          },
+                          [&seed](const op::Convert& op) {
+                              seed ^= op.type.hash() + 0x9e3779b9;
+                              seed ^= op.tensor.get_hash() + 0x9e3779b9;
+                          },
+                          [&seed](const op::Concat& op) {
+                              seed ^= std::hash<std::size_t>()(op.axis) + 0x9e3779b9;
+                              for (auto& lt : op.tensors) {
+                                  seed ^= lt.get_hash() + 0x9e3779b9;
+                              }
+                          },
+                          [&seed](const op::Unpack& op) {
+                              seed ^= op.w.get_hash() + 0x9e3779b9;
+                              seed ^= op.z.get_hash() + 0x9e3779b9;
+                              seed ^= op.s.get_hash() + 0x9e3779b9;
+                              seed ^= op.type.hash() + 0x9e3779b9;
+                              for (const auto& dim : op.shape) {
+                                  seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
+                              }
+                          }},
+               m_transform);
 
     return seed;
 }
-}  // namespace weights
-}  // namespace npuw
-}  // namespace ov
 
-using ov::npuw::weights::LazyTensorImpl;
+const void* LazyTensorImpl::get_data() const {
+    return m_orig_data;
+}
```

Comment:
What's the point, if there's no orig_data?

Reply:
Removed